### PR TITLE
add METHOD convert_to_table.

### PR DESCRIPTION
**Pull request by boy0korea**
METHOD convert_to_table.
* For excel upload, This is counterpart to BIND_TABLE
* This method has 2 exporting parameters.
* ER_DATA is a reference to internal table of string columns. intend to get raw data without conversion loss.
* ET_DATA is an internal table supplied. There may be losses in the conversion(MOVE-CORRESPONDING).

**Request** from: https://github.com/boy0korea/ABAP2XLSX_HELPER/issues/1#issue-1144673227

### CODE REVIEW

**Comment by AndreaBorgia-Abo** [comment]
Before I begin the review, one quick question: have you noticed that now the demos and the main library are split into two distinct repositories? We would need a demo for this new method and that demo goes into the new ["demos" repository](https://github.com/abap2xlsx/demos).

**Comment by AndreaBorgia-Abo** [comment]
@boy0korea do you have time for the demo? No pressure, just want to know if I should take care of that.

**Comment by larshp** [comment]
1 issue from linter: is https://github.com/abap2xlsx/abap2xlsx/pull/1056 waiting to be merged

**Comment by AndreaBorgia-Abo** [comment]
@boy0korea could you please look at the linter messages?
Also, what about the demo program?

**Comment by boy0korea** [comment]
> @boy0korea could you please look at the linter messages? Also, what about the demo program?

demo: https://github.com/abap2xlsx/demos/pull/23

File: src/zcl_excel_worksheet.clas.abap [hunk 1]
```diff
@@ -664,6 +664,15 @@ CLASS zcl_excel_worksheet DEFINITION
         !ip_is_internal  TYPE abap_bool
       RAISING
         zcx_excel .
+    METHODS convert_to_table
```
Comment (AndreaBorgia-Abo) [inline src/zcl_excel_worksheet.clas.abap:672]:
```suggestion
    "! excel upload, counterpart to BIND_TABLE
    "! @parameter it_field_catalog | field catalog, used to derive correct types
    "! @parameter iv_begin_row | starting row, by default 2 to skip header
    "! @parameter et_data | generic internal table, there may be conversion losses
    "! @parameter er_data | ref to internal table of string columns, to get raw data without conversion losses.
    METHODS convert_to_table
```

Comment (AndreaBorgia-Abo) [reply src/zcl_excel_worksheet.clas.abap:672]: @larshp I'm not entirely sure how back in time we can go with this comment style, but I'd like it better this way if at all possible. EDIT: never mind, I guess the linter doesn't really care about what is in a comment

File: src/zcl_excel_worksheet.clas.abap [hunk 2]
```diff
@@ -2051,6 +2060,335 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
   ENDMETHOD.                    "CONSTRUCTOR
 
 
+  METHOD convert_to_table.
+* For excel upload, This is counterpart to BIND_TABLE
+* This method has 2 exporting parameters.
+* ER_DATA is a reference to internal table of string columns. intend to get raw data without conversion loss.
+* ET_DATA is an internal table supplied. There may be losses in the conversion(MOVE-CORRESPONDING).
```
Comment (AndreaBorgia-Abo) [inline src/zcl_excel_worksheet.clas.abap]:
```suggestion

```

Comment (AndreaBorgia-Abo) [reply src/zcl_excel_worksheet.clas.abap]: Goes together with earlier suggestion, ignore if the one above is rejected

File: src/zcl_excel_worksheet.clas.abap [hunk 3]
```diff
@@ -2051,6 +2060,335 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
...
+*                READ TABLE lt_map_excel_row INTO lv_index INDEX lv_index.
+*                READ TABLE me->sheet_content ASSIGNING <ls_sheet_content> WITH TABLE KEY cell_row = lv_index cell_column = lv_index_col.
+*                ls_error_log-excel_coords = <ls_sheet_content>-cell_coords.
+*                APPEND ls_error_log TO et_error_log.
+*              ENDIF.
+*            ENDLOOP.
+*          ENDIF.
+*        ENDLOOP.
+*      ENDIF.
+*    ENDIF.
```
Comment (AndreaBorgia-Abo) [inline src/zcl_excel_worksheet.clas.abap]:
```suggestion

```

**Review by AndreaBorgia-Abo — CHANGES_REQUESTED** [review]
Have a look at the suggestions, no big deal since the code works with the demo.
I'll fix the linter error in a moment.
Good job and, once again, welcome on board!

**Comment by AndreaBorgia-Abo** [comment]
> 1 issue from linter: is #1056 waiting to be merged

How do we merge this? It's in another branch which requires this to be merged first which requires the linter to be happy. I'd simply fix the issue in the PR branch and be on my way.

**Pull request merged**